### PR TITLE
Add dry run functionality for stop start restart commands

### PR DESCRIPTION
**Pull request by fossedihelm**
**What this PR does / why we need it**:
Allows user to use `--dry-run` flag for start, stop and restart virtctl command.
Passing this flag to one of these command will do all checks showing possible failures, but changes will not be effective.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Users can use --dry-run flag
```

### CODE REVIEW

**Review by jean-edouard — COMMENTED** [review]
/approve
Couple nits in-line, probably not worth fixing on their own if nothing else pops up though!

File: staging/src/kubevirt.io/client-go/apis/core/v1/types.go [hunk 1]
```diff
@@ -1840,6 +1849,14 @@ type StartOptions struct {
...
 	// +optional
 	Paused bool `json:"paused,omitempty" protobuf:"varint,7,opt,name=paused"`
+	// When present, indicates that modifications should not be
+	// persisted. An invalid or unrecognized dryRun directive will
+	// result in an error response and no further processing of the
+	// request. Valid values are:
+	// - All: all dry run stages will be processed
+	// +optional
+	// +listType=atomic
+	DryRun []string `json:"dryRun,omitempty" protobuf:"bytes,5,rep,name=dryRun"`
```
Comment (jean-edouard) [inline staging/src/kubevirt.io/client-go/apis/core/v1/types.go:1859]: Could you please explain the protobuf fields here, and why the second value is 5 on this one and 2 on the others?
(Feel free to just point me to the documentation, I just couldn't find it!)

Comment (jean-edouard) [reply staging/src/kubevirt.io/client-go/apis/core/v1/types.go:1859]: /hold
Feel free to unhold, it just makes sense to keep this open while there's an open question.

File: pkg/virtctl/vm/vm.go [hunk 2]
```diff
@@ -54,11 +54,12 @@ const (
 
 var (
 	forceRestart bool
-	gracePeriod  int = -1
+	gracePeriod  int64 = -1
```
[the remark this answers is not in the document]
Comment (fossedihelm) [reply pkg/virtctl/vm/vm.go:57]: Thanks. I changed the default value at a later time and forgot to remove the initialization. Will fix

File: pkg/virtctl/vm/vm.go [hunk 3]
```diff
@@ -89,8 +91,9 @@ func NewStopCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
 		},
 	}
 
-	cmd.Flags().BoolVar(&forceRestart, "force", forceRestart, "--force=false: Only used when grace-period=0. If true, immediately remove VMI pod from API and bypass graceful deletion. Note that immediate deletion of some resources may result in inconsistency or data loss and requires confirmation.")
-	cmd.Flags().IntVar(&gracePeriod, "grace-period", gracePeriod, "--grace-period=-1: Period of time in seconds given to the VMI to terminate gracefully. Can only be set to 0 when --force is true (force deletion). Currently only setting 0 is supported.")
+	cmd.Flags().BoolVar(&forceRestart, "force", false, "--force=false: Only used when grace-period=0. If true, immediately remove VMI pod from API and bypass graceful deletion. Note that immediate deletion of some resources may result in inconsistency or data loss and requires confirmation.")
+	cmd.Flags().Int64Var(&gracePeriod, "grace-period", -1, "--grace-period=-1: Period of time in seconds given to the VMI to terminate gracefully. Can only be set to 0 when --force is true (force deletion). Currently only setting 0 is supported.")
```
[the remark this answers is not in the document]
Comment (fossedihelm) [reply pkg/virtctl/vm/vm.go:95]: Nope, I didn't seen that until you point me out. Will fix, thanks

File: staging/src/kubevirt.io/client-go/apis/core/v1/types.go (hunk 1 above)
Comment (fossedihelm) [reply staging/src/kubevirt.io/client-go/apis/core/v1/types.go:1859]: I think it was just a mistake. Here's the [documentation](https://developers.google.com/protocol-buffers/docs/overview#assigning_field_numbers). If I understand correctly the second field basically is a unique identifier "_used to identify your fields in the message binary format_".

Comment (jean-edouard) [reply staging/src/kubevirt.io/client-go/apis/core/v1/types.go:1859]: /unhold

**Pull request merged**